affinidi-messaging-mediator/src/messages/protocols/message_pickup.rs

```diff
@@ -331,6 +331,26 @@ pub(crate) async fn delivery_request(
                 }
             };
 
+        if session.did != recipient_did {
```

Comment:
This method is quite large. It would be beneficial if this and future contributions could break it down into smaller, more manageable methods.   
Could the part related to extracting input values and validating them be moved to a separate method? This would help with clarity and separation of concerns.